### PR TITLE
generic temp and humidity calibration

### PR DESCRIPTION
**Pull request by stif**
Hi Cristian,

thank you for merging my PR and great idea to put the Device Selection right to the make.sh file.

there are some small problems with your subsequent modifications tough:
The calibration was meant as a general offset for all Temp/Humdity sensors, although i just implemented it for the SHT31 and BME280 Sensor. So there is still a uninitialized `TemperatureOffset` Variable inside the `#ifdef _BME280_H` Block:
https://github.com/cbalint13/tinnymodbus/blob/a995924d8ee5eb4d630d51ad6e8c77318fd6ccef/main.c#L665 
Another small issue is the `#endif` after the switch break in [main.c Line 371](https://github.com/cbalint13/tinnymodbus/blob/a995924d8ee5eb4d630d51ad6e8c77318fd6ccef/main.c#L371), so the break is not included when SHT31 is not defined.
In my new Pull Request i corrected these minor issues and made the calibration generic for all temp/humidity sensors except the DS18B20 (there could be more than one and i have only one calibration value..).
I also added updated .hex and .eep files to reflect the EEPROM is changed from a single byte to an array of bytes.

Cheers,
Stif

### CODE REVIEW

File: boot.eep
Comment (cbalint13) [inline boot.eep:1]: * let's remove pregenerated things like .eep
* users can generate their own in accord with their needs

File: boot.hex
Comment (cbalint13) [inline boot.hex:1]: * remove, same as .eep

File: main.c [hunk 1]
```diff
@@ -144,11 +144,11 @@ int main(void)
     // fetch own slave address from EEPROM
     uint8_t IdSv = eeprom_read_byte(&EEData[0]);
 
-    #ifdef _SHT31_H
+    #ifdef CALIBRATION
     // fetch Temperature Offset from EEPROM
-    float SHT31_TOffset = eeprom_read_byte(&EEData[1]) / 10.0f;
+    float generic_Temp_Offset = eeprom_read_byte(&EEData[1]) / 10.0f;
```
Comment (cbalint13) [inline main.c:149]: * just a cosmetic nit (not mandatory)
* s/generic_TempOffset/CAL_TOffset/

File: make.sh [hunk 2]
```diff
@@ -9,15 +9,17 @@ rm -rf *.eep
 ## Enabled DEVICES
 ##
 ##   DS18B20  (1w    | temperature)
-##   SHT21    (i2c   | temperature + humidity)
-##   SHT31    (i2c   | temperature + humidity + calbibration)
+##   SHT21    (i2c   | temperature + humidity + optional calbibration)
```
Comment (cbalint13) [inline make.sh:12]: small typo: "calibration"

**Review by cbalint13 — CHANGES_REQUESTED** [review]
@stif ,
Thanks for the extras, it is fine just few nits needing requested changes (cosmetic) !

File: make.sh (hunk 2 above)
Comment (stif) [reply make.sh]: i removed the .hex and .eep files and did fixed the typos 👍

File: main.c (hunk 1 above)
Comment (stif) [reply main.c]: renamed the calibration variables

**Comment by cbalint13** [comment]
@stif ,
Thank you Stefan !

**Comment by stif** [comment]
thank you for the quick merge! 🤝

**Pull request merged**